### ISSUE TITLE
Remove empty files `SystemInformer.exe.settings.xml` and `usernotesdb.xml`

### ISSUE THREAD

**Issue by stdedos** [winsiderss/systeminformer#1585]
### Description of the feature, modification, idea or suggestion

For the few "manual upgrade" (or downgrade) operation, it'd be nice to not have these files, and SI would just `touch` them if necessary

**Comment by dmex** [comment]
`SystemInformer.exe.settings.xml` can not be deleted because if it even exists then it is the current settings file.

`usernotesdb.xml` is already deleted when it's not used or empty:

https://github.com/winsiderss/systeminformer/blob/bbc7d0932d4cb3249aa334449a0bd7c3abf1613b/plugins/UserNotes/db.c#L323-L328
https://github.com/winsiderss/systeminformer/blob/bbc7d0932d4cb3249aa334449a0bd7c3abf1613b/plugins/UserNotes/db.c#L377-L385

**Comment by stdedos** [comment]
Nonono 😅 

The zip artifacts contain those empty files. A blind extract on a "running" instance of SI will remove those files.

It is PEBKAC ofc - I am just trying to save some absent-minded people from shooting themselves 😓

**Comment by dmex** [comment]
> The zip artifacts contain those empty files.

The zip is supposed to contain both files since the zip is for portable installation?

**Comment by stdedos** [comment]
ergo - that's how SI understands it's in portable mode?

If so, could there be another mechanism? Like an empty file called `portable` or something, which will carry no data?

**Comment by dmex** [comment]
> that's how SI understands it's in portable mode?

Yes

> If so, could there be another mechanism? Like an empty file called portable or something, which will carry no data?

What benefit would there be from using multiple files? They would still have to be added to the zip file.

If you extract the zip into a folder and run from the folder then everything is stored in the folder? The zip is for portable so delete the folder and there's nothing left on the system. You can run multiple versions side-by-side (stable/nightly) without worrying about one corrupting the other and since portable mode must be checked during startup this design is high performance because it only requires a single file query.

**Comment by stdedos** [comment]
There's a slight issue I've bumped into:

When SI is selected to replace Task Manager, it seems that "no other instance" can start - be it TaskManager, other portable SI/PH, etc. 😕 

> What benefit would there be from using multiple files?

The portable-indicator is a dummy file that serves this job. "Replacing" it causes no damage, as it never contains anything.
The settings only stores the Settings. It's not in the archive, so it cannot be accidentally replaced with an empty file.

tl;dr this sequence is not a disaster:
![image](https://user-images.githubusercontent.com/133706/215585399-975f5e3f-437a-4fe9-b5ea-8fd06356359c.png)
![image](https://user-images.githubusercontent.com/133706/215585668-2f2f73d5-264b-4891-b5bd-97d443badc5f.png)

**Comment by dmex** [comment]
> When SI is selected to replace Task Manager, it seems that "no other instance" can start

 I've added a warning when changing the path for portable builds. There's not much else that can be done about this considering Microsoft's anti-competitive behavior prohibiting changes to the users default application. This is only a problem for Task Manager since APIs exist for the other defaults.

> The portable-indicator is a dummy file that serves this job.

That's what the file currently does.

> "Replacing" it causes no damage, as it never contains anything. The settings only stores the Settings. It's not in the archive, so it cannot be accidentally replaced with an empty file.

These paths have to be checked on startup every time you start the application. The added complexity of requiring multiple files and checking multiple files (combined with AV filters delaying file access) slows startup. The zip still has to be portable by default when extracted so it would still need these extra files and then you're back to the square one just with more files.
